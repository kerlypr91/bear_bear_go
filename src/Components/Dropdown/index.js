import React from 'react'
import { Dropdown as SemanticDropdown } from 'semantic-ui-react'

export default function Dropdown (props) {
  const { placeholder, options, className, onChange, value } = props

  const handleChange = (event, data) => {
    const { value } = data

    event.preventDefault()

    if (!onChange) {
      return
    }

    onChange(value)
  }

  return (
    <SemanticDropdown
      placeholder={placeholder}
      selection
      options={options.map(item => {
        return {
          key: item.toLowerCase(),
          text: item,
          value: item.toLowerCase()
        }
      })}
      onChange={handleChange}
      value={value}
      className={className || ''}
    />
  )
}