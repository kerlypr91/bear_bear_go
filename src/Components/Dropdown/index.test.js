import React from 'react';
import renderer from 'react-test-renderer';
import Dropdown from './index';

test('Dropdown match snapshot', () => {
  const defaultSpy = jest.fn()
  const onChangeSpy = jest.fn()

  const props = {
    placeholder: 'test',
    options: ['a', 'b'],
    className: '',
    onChange: onChangeSpy,
    value: 'a'
  }

  const component = renderer.create(
    <Dropdown {...props} />
  )

  // base snapshot

  let tree = component.toJSON()

  expect(tree).toMatchSnapshot()

  // onChange prop gets called when a different dropdown value is selected

  tree.props.onChange({ preventDefault: defaultSpy }, 'b')

  expect(defaultSpy).toHaveBeenCalled()
  expect(onChangeSpy).toHaveBeenCalledWith('b')
});