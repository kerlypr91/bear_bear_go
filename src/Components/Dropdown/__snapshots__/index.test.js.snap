// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown match snapshot 1`] = `
<div
  aria-expanded={false}
  className="ui selection dropdown"
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  role="listbox"
  tabIndex={0}
>
  <div
    aria-atomic={true}
    aria-live="polite"
    className="divider text"
    role="alert"
  >
    a
  </div>
  <i
    aria-hidden="true"
    className="dropdown icon"
    onClick={[Function]}
  />
  <div
    className="menu transition"
  >
    <div
      aria-checked={true}
      aria-selected={true}
      className="active selected item"
      onClick={[Function]}
      role="option"
      style={
        Object {
          "pointerEvents": "all",
        }
      }
    >
      <span
        className="text"
      >
        a
      </span>
    </div>
    <div
      aria-checked={false}
      aria-selected={false}
      className="item"
      onClick={[Function]}
      role="option"
      style={
        Object {
          "pointerEvents": "all",
        }
      }
    >
      <span
        className="text"
      >
        b
      </span>
    </div>
  </div>
</div>
`;
